Sprint 4

tikhomi13 opened the pull request:
1.  test_set_book_genre_add_invalid_genres()

Исправил. Задействовал в assert метод get_book_genre, который возвращает пустую строку, если передаем невалидные жанры в set_book_genre 

2.  test_set_book_genre_only_valid_items_in_genre_list()

Исправил методы в ассертах на те, что возвращают данные, + разделил ожидаемые результаты на None и Пустую строку, в зависимости от входящих данных 

3. test_get_books_genre_add_three_books_to_genre_list() 

Оставил тест, так как метод ранее не проверялся. Проверялся метод с похожим названием get_book_genre (не get_bookS_ genre). Оставил тест, так как в задании сказано покрыть каждый метод. 

Переименовал данный метод в test_get_books_genre_add_three_books_to_genre_list и сделал парметризованным, чтобы он проверял точное содержание набора значений, добавленных в словарь books_genre. 

4.  test_add_book_in_favorites_add_book_to_favorites()
5.  test_list_of_favorites_books_request_favorites

Исправление пунктов 4; 5. Запись в переменную значений, возваращаемых методом get_list_of_favorites_books

tikhomi13 committed:
README Changed

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,129 @@
+
+        book.add_new_book(test_input)
+
+        assert book.set_book_genre('Тарзан', test_input) == None
```
Нужно исправить: этот метод ничего не возвращает, он не может использоваться в assert

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,129 @@
+
+        book.add_new_book(invalid_genre)
+
+        assert book.set_book_genre(no_book_in_genre, 'Ужасы') or book.add_new_book(invalid_genre) == None
```
Нужно исправить: оба этих метода ничего не возвращают, и не могут использоваться в assert

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,129 @@
+
+        assert len(book.get_books_with_specific_genre('Мультфильмы')) == 2
+
+    def test_get_books_genre_add_three_books_to_genre_list(self):
+
+        book = BooksCollector()
+        book.add_new_book('Тестирование и сон')
+        book.add_new_book('Автоматизация')
+        book.add_new_book('Pytest')
+
+        book.set_book_genre('Тестирование и сон', 'Фантастика')
+        book.set_book_genre('Автоматизация', 'Ужасы')
+        book.set_book_genre('Pytest', 'Детективы')
+
+        assert len(book.get_books_genre()) == 3
```
Уже проверено выше

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,129 @@
+        book = BooksCollector()
+        book.add_new_book('Любимая книга')
+        book.add_book_in_favorites('Любимая книга')
+        book.get_list_of_favorites_books()
```
Нужно исправить: этот метод возвращает значение, но они никуда не записывается

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,129 @@
+        book = BooksCollector()
+        book.add_new_book('Fav_book')
+        book.add_book_in_favorites('Fav_book')
+        book.get_list_of_favorites_books()
```
up

tikhomi13 committed:
README Changed

tikhomi13 committed:
1.  test_set_book_genre_add_invalid_genres()

Исправил. Задействовал в assert метод get_book_genre, который возвращает пустую строку, если передаем невалидные жанры в set_book_genre

2.  test_set_book_genre_only_valid_items_in_genre_list()

Исправил методы в ассертах на те, что возвращают данные, + разделил ожидаемые результаты на None и Пустую строку, в зависимости от входящих данных

3. test_get_books_genre_add_three_books_to_genre_list()

Оставил тест, так как метод ранее не проверялся. Проверялся метод с похожим названием get_book_genre (не get_bookS_ genre), а в задании сказано покрыть тестами каждый метод.

Переименовал данный метод в test_get_books_genre_add_three_books_to_genre_list и сделал парметризованным, чтобы проверить, точное содержание набора значений, добавленных в словарь books_genre.

4.  test_add_book_in_favorites_add_book_to_favorites()

Не понял, что тут менять. В методе создается  объект "Любимая книга". Затем книга добавляется в список favorites с помощью метода add_book_in_favorites. Ассерт проверяет, что значение 'Любимая книга' действительно записалось в данный список.

В комментарии сказано, что значение никуда не записывается.
Добавил проверку, что книга также содержится в словаре books_genre. Не представляю, что здесь еще можно сделать. Если действительно что-то не так  -  готов доработать проект еще раз.

5.  test_list_of_favorites_books_request_favorites()

Теперь метод проверяет длину словаря favorites

tikhomi13 committed:
Update tests.py

исправление четвертого замечения. Запись в переменную значения, возваращаемого методом get_list_of_favorites_books

tikhomi13 committed:
Update tests.py

Исправление пункутов 4; 5. Запись в переменную значений, возваращаемых методом get_list_of_favorites_books

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,130 @@
+            book.set_book_genre('Автоматизация', 'Ужасы')
+            book.set_book_genre('Pytest', 'Детективы')
+
+            assert check_values_in_books_genre_dict in book.get_books_genre().values()
```
in get_books_genre(), values() тут зачем?
